bpdm-cleaning-service-dummy/src/main/kotlin/org/eclipse/tractusx/bpdm/cleaning/service/CleaningServiceDummy.kt

```diff
@@ -0,0 +1,146 @@
+            val cleaningRequest = orchestrationApiClient.goldenRecordTasks
+                .reserveTasksForStep(TaskStepReservationRequest(amount = 10, TaskStep.Clean))
+
+            val cleaningTasks = cleaningRequest.reservedTasks
+
+            logger.info { "${cleaningTasks.size} tasks found for cleaning. Proceeding with cleaning..." }
+
+            if (cleaningTasks.isNotEmpty()) {
+
+                val cleaningResults = cleaningTasks.mapNotNull { reservedTask ->
+                    // Step 2: Generate dummy cleaning results
+                    processCleaningTask(reservedTask)
+                }
+
+                // Step 3: Send the cleaning result back to the Orchestrator
+                orchestrationApiClient.goldenRecordTasks.resolveStepResults(TaskStepResultRequest(cleaningResults))
+                logger.info { "Cleaning tasks processing completed for this iteration." }
+            }
+        } catch (e: Exception) {
+            logger.error(e) { "Error while processing cleaning task" }
+        }
+    }
+
+    fun processCleaningTask(reservedTask: TaskStepReservationEntryDto): TaskStepResultEntryDto {
+        val businessPartner = reservedTask.businessPartner
+
+        val addressPartner = createAddressRepresentation(businessPartner.generic)
+
+        val addressType = businessPartner.generic.postalAddress.addressType
+
+        val legalEntityDto = createLegalEntityRepresentation(addressPartner, addressType!!, businessPartner.generic)
+
+        val siteDto = createSiteDtoIfNeeded(businessPartner.generic, addressPartner)
+
+        return TaskStepResultEntryDto(reservedTask.taskId, BusinessPartnerFullDto(businessPartner.generic, legalEntityDto, siteDto, addressPartner))
+    }
+
+    fun createSiteDtoIfNeeded(businessPartner: BusinessPartnerGenericDto, addressPartner: LogisticAddressDto): SiteDto? {
+        if (!shouldCreateSite(businessPartner)) return null
+
+        val siteAddressReference = when (businessPartner.postalAddress.addressType) {
+            AddressType.SiteMainAddress, AddressType.LegalAndSiteMainAddress -> addressPartner.bpnAReference
+            else -> generateNewBpnReference()
+        }
+
+        val siteMainAddress = addressPartner.copy(bpnAReference = siteAddressReference)
+        return createSiteRepresentation(businessPartner, siteMainAddress)
+    }
+
+    fun createLegalEntityRepresentation(
+        addressPartner: LogisticAddressDto,
+        addressType: AddressType,
+        genericPartner: BusinessPartnerGenericDto
+    ): LegalEntityDto {
+        val legalAddressBpnReference = if (addressType == AddressType.LegalAddress || addressType == AddressType.LegalAndSiteMainAddress) {
+            addressPartner.bpnAReference
+        } else {
+            generateNewBpnReference()
+        }
+
+        val legalAddress = addressPartner.copy(bpnAReference = legalAddressBpnReference)
+
+        val legalName = genericPartner.nameParts.joinToString(" ")
```

Comment:
Why not calculating the `legalName` inside`BusinessPartnerGenericDto.toLegalEntityDto()` ?